src/coreclr/nativeaot/Runtime/gcrhenv.cpp

```diff
@@ -1367,26 +1367,29 @@ bool GCToEEInterface::GetBooleanConfigValue(const char* privateKey, const char*
     }
 
     uint64_t uiValue;
-    if (!g_pRhConfig->ReadConfigValue(privateKey, &uiValue))
-        return false;
+    if (g_pRhConfig->ReadConfigValue(privateKey, &uiValue))
+    {
+        *value = uiValue != 0;
+        return true;
+    }
 
-    *value = uiValue != 0;
-    return true;
+    if (publicKey)
+    {
+        if (g_pRhConfig->ReadKnobValue(publicKey, &uiValue))
+        {
+            *value = uiValue != 0;
+            return true;
+        }
```

Comment:
This won't work for settings that specify `true`/`false`. Could you mirror how this is handled in CoreCLR?